prune.go

```diff
@@ -100,6 +100,41 @@ func shouldPrune(oldCount, newCount int, pruneOptions PruneOptions) (bool, strin
 	return false, "transactions have not grown significantly"
 }
 
+// checkMongoSupportsOut verifies that Mongo supports "$out" in an aggregation
+// pipeline. This was introduced in Mongo 2.6
+// https://docs.mongodb.com/manual/reference/operator/aggregation/out/
+func checkMongoSupportsOut(db *mgo.Database) bool {
+	var dbInfo struct {
+		VersionArray []int `bson:"versionArray"`
+	}
+	if err := db.Run(bson.M{"buildInfo": 1}, &dbInfo); err != nil {
+		return false
+	}
+	logger.Debugf("buildInfo reported: %v", dbInfo.VersionArray)
+	if len(dbInfo.VersionArray) < 2 {
+		return false
+	}
+	// Check if we are < 2.6
+	if dbInfo.VersionArray[0] < 2 ||
+		(dbInfo.VersionArray[0] == 2 && dbInfo.VersionArray[1] < 6) {
+		return false
+	}
+	return true
+}
```

Comment:
Done. I also realized that because of not being able to put the pre-check in SetUpTest() it was skipping memory tests as well. So I factored that into a PreCheck that is different for memory and DB, and I added a test to the DBOracleSuite that checks that Pipe returns a known error when checkMongo returns false, and returns no error when it doesn't.